### PR TITLE
GH-1266 Remove redundant cpu() and detach() calls

### PR DESCRIPTION
**Pull request by shoarora**
#1266 

Currently we call `.cpu()` and `detach()` in for loop on a per-token basis.  AFAIK, this is redundant because when we assign the embedding to the token, we will allocate it to the right device/detach it as necessary.

As I outlined in the issue and the associated [gist](https://gist.github.com/shoarora/a8aa6ac39ee814b667907ced4bfda6c9), this will dramatically improve runtime, as we'll move to `cpu()` _after_ we've aggregated all the desired embeddings.  

I didn't receive confirmation on my issue that this functionality works the way that I understand it.  Hopefully, I'm not missing a key piece of the implementation that actually makes this required.

### CODE REVIEW

**Comment by alanakbik** [comment]
@shoarora thanks for spotting and fixing this and providing the runtime tests! Also please excuse the late reply!

Since everything is called inside `with torch.no_grad():` it makes sense that the detach is not necessary. @stefan-it do you agree? I notice also that the lines 
```python 
self.model.to(flair.device)
self.model.eval() 
```
are called each time `_add_embeddings_internal` is called - is this necessary? Isn't this already handled by `torch.nn.Module`?

**Comment by alanakbik** [comment]
@shoarora again thanks for fixing this and excuse the long time to review - in my local tests this lets BERT run dramatically faster! Merging now!

**Comment by alanakbik** [comment]
:+1:
